src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,243 @@
+        var _initTL: Point;
+        var _initBR: Point;
+
+        this._dragInteraction.onDragStart((s: Point) => {
+          grabbedEdges = this._getEdges(s);
+          if (!this._yResizable) {
+            grabbedEdges.top = false;
+            grabbedEdges.bottom = false;
+          }
+          if (!this._xResizable) {
+            grabbedEdges.left = false;
+            grabbedEdges.right = false;
+          }
+
+          if (!this.boxVisible() ||
+              (!grabbedEdges.top && !grabbedEdges.bottom &&
+               !grabbedEdges.left && !grabbedEdges.right)
+             ) {
+            this.bounds({
+              topLeft: s,
+              bottomRight: s
+            });
+            // a new box is effectively "grabbed" at the bottom-right corner
+            grabbedEdges.bottom = true;
+            grabbedEdges.right = true;
+          }
+
+          this.boxVisible(true);
+          var bounds = this.bounds();
+          _initTL = bounds.topLeft;
+          _initBR = bounds.bottomRight;
+        });
+
+        this._dragInteraction.onDrag((s: Point, e: Point) => {
+          var topLeft = _initTL;
+          var bottomRight = _initBR;
+
+          if (grabbedEdges.bottom) {
+            bottomRight.y = e.y;
+          } else if (grabbedEdges.top) {
+            topLeft.y = e.y;
+          }
+
+          if (grabbedEdges.right) {
+            bottomRight.x = e.x;
+          } else if (grabbedEdges.left) {
+            topLeft.x = e.x;
+          }
+
+          this.bounds({
+            topLeft: topLeft,
+            bottomRight: bottomRight
+          });
+        });
+
+        this._dragInteraction.onDragEnd((s: Point, e: Point) => {
+          if (s.x === e.x && s.y === e.y) {
+            this.boxVisible(false);
+          }
+        });
+
+        this.registerInteraction(this._dragInteraction);
+      }
+
+      protected _setup() {
+        super._setup();
+
+        var createLine = () => this._box.append("line").style({
+                                 "opacity": 0,
+                                 "stroke": "pink"
+                               });
+        this._boxEdgeT = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeB = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeL = createLine().classed("drag-edge-lr", true);
+        this._boxEdgeR = createLine().classed("drag-edge-lr", true);
+
+        var createCorner = () => this._box.append("circle")
+                                     .style({
+                                       "opacity": 0,
+                                       "fill": "pink"
+                                     });
+        this._boxCornerTL = createCorner().classed("drag-corner-tl", true);
+        this._boxCornerTR = createCorner().classed("drag-corner-tr", true);
+        this._boxCornerBL = createCorner().classed("drag-corner-bl", true);
+        this._boxCornerBR = createCorner().classed("drag-corner-br", true);
```

Comment:
Or you could instead say

if either of xResize or yResize is false, skip this step

(or if both are true, then do this step)